db.go

```diff
@@ -249,3 +263,12 @@ func readEntryLengths(reader *bufio.Reader) (int, int) {
 	valLen, _ := binary.ReadUvarint(reader)
 	return int(keyLen), int(valLen)
 }
+
+func (db *DB) OpenSSTableByFileName(fileName string) (*SSTable, error) {
+	file, err := os.Open(fileName)
```

Comment:
### Security Issues

- **File Path Injection**: Ensure that the `fileName` passed to `OpenSSTableByFileName` is sanitized to prevent file path injection attacks. Validate that the file path is within the expected directory and does not contain any unexpected characters or patterns.

```go
if !isValidFilePath(fileName) {
    return nil, fmt.Errorf("invalid file path")
}
```